DanBloomberg/leptonica
# improve testing support and the CI

Testing:
- fix testing on macOS: ensure the same temporary directory is used by redirecting all `/tmp/…` paths accesses to the same location (similarly to was is done on Windows)
- add testing support to the cmake build
- add support for running the tests in parallel: isolate each test temporary directory

CI:
- fix the `cmake_win64` workflow: the libtiff build was failing (disable LZMA support for now), the version of giflib being built was 5.1.2 (broken)
- enable the `cmake_win64` workflow for all branches
- stop a workflow previous run when a new one is started (on the same branch/tag)
- enable the tests on the `cmake_…` workflows
